tsl/src/compression/compression.h

```diff
@@ -73,13 +73,16 @@ struct Compressor
 	void *(*finish)(Compressor *data);
 };
 
+typedef struct ArrowArray ArrowArray;
+
 typedef struct DecompressionIterator
 {
 	uint8 compression_algorithm;
 	bool forward;
 
 	Oid element_type;
 	DecompressResult (*try_next)(struct DecompressionIterator *);
+	ArrowArray (*decompress_all)(struct DecompressionIterator *);
```

Comment:
Do you want to return the ArrowArray struct or a pointer to a palloc:ed structure?

```suggestion
	ArrowArray *(*decompress_all)(struct DecompressionIterator *);
```

Reply:
This was a leftover from old code, removed.